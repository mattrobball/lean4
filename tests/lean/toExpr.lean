import Lean

open Lean BitVec

open Meta in
def test [ToExpr α] (a : α) : MetaM Unit := do
  let e := toExpr a
  let type ← inferType e
  check e
  logInfo m!"{e} : {type}"

run_meta test (2 : Fin 4)
run_meta test (8 : Fin 5)
run_meta test (300#8)
run_meta test (300#32)
run_meta test (58#8)
run_meta test (20 : UInt8)
run_meta test (30 : UInt16)
run_meta test (40 : UInt32)
run_meta test (50 : UInt64)
run_meta test (60 : USize)
