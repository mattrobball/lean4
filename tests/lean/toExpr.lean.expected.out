2 : Fin 4
3 : Fin 5
44#8 : BitVec 8
300#32 : BitVec 32
58#8 : BitVec 8
20 : UInt8
30 : UInt16
40 : UInt32
50 : UInt64
60 : USize
